mapic/stats_collector.go

```diff
@@ -30,45 +31,27 @@ type metricsCollector struct {
 	infoProvider
 }
 
-func startMetricsCollector(ctx context.Context, period time.Duration, nodeID, ownRegion string, mapi clients.MistAPIClient, lapi *api.Client, producer event.AMQPProducer, amqpExchange string, infop infoProvider) {
+func createMetricsCollector(ctx context.Context, period time.Duration, nodeID, ownRegion string, mapi clients.MistAPIClient, lapi *api.Client, producer event.AMQPProducer, amqpExchange string, infop infoProvider) *metricsCollector {
```

Comment:
We don't need this `ctx` argument anymore, that one was only used in the `mainLoop`. `period` is also unused. Make sure to remove any others.